distributed/tests/test_executor.py

```diff
@@ -1051,6 +1051,78 @@ def test_directed_scatter_sync(loop):
             assert len(has_what[('127.0.0.1', a['port'])]) == 0
 
 
+def test_iterator_scatter(loop):
+    with cluster() as (s, [a, b]):
+        with Executor(('127.0.0.1', s['port']), loop=loop) as ee:
+
+            aa = ee.scatter([1,2,3])
+            assert [1,2,3] == ee.gather(aa)
+
+            g = (i for i in range(10))
+            futures = ee.scatter(g)
+            assert isinstance(futures, Iterator)
+
+            a = next(futures)
+            assert ee.gather(a) == 0
+
+
+def test_queue_scatter(loop):
+    with cluster() as (s, [a, b]):
+        with Executor(('127.0.0.1', s['port']), loop=loop) as ee:
+            from distributed.compatibility import Queue
+            q = Queue()
+            [q.put(d) for d in range(10)]
```

Comment:
Generally prefer for loops for imperative actions rather than list comprehensions 

(This point is not important.  Feel free to ignore.)